### PR TITLE
drivers: pwm: Add Intel PCH blinky PWM driver

### PR DESCRIPTION
**Pull request by akanisetti**
Submitting this PR for review for new PWM driver for the IP found in multiple intel PCH hardwares.
This PR includes commits for
a) The driver code
b) dts node for pwm
c) enable pwm_api test for rpl_crb

we can verify using "west build -b rpl_crb tests/drivers/pwm/pwm_api/"

### CODE REVIEW

File: drivers/pwm/pwm_intel_blinky.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			       pwm_flags_t flags)
+{
+	uint32_t ret = 0;
+	uint32_t val = 0;
+	uint32_t duty;
+	float period;
+	float out_freq;
+	uint32_t base_unit;
+	struct bk_intel_runtime *rt = dev->data;
+	const struct bk_intel_config *cfg = dev->config;
```
Comment (finikorg) [inline drivers/pwm/pwm_intel_blinky.c:46]: I think it is better to move those 2 declarations above to the beginning of the declaration block.

File: drivers/pwm/Kconfig.intel_blinky [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+config PWM_INTEL_BLINKY
+	bool "Blinky PWM driver"
+	default y
+	depends on DT_HAS_INTEL_BLINKY_PWM_ENABLED
+	help
+	  Enable the INTEL PCH PWM driver found on Intel SoCs
+
+config PWM_INTEL_CONTROL_OFFSET
+	int "PWM intel control register offset"
```
Comment (finikorg) [inline drivers/pwm/Kconfig.intel_blinky:14]: Can we use `hex` for offset?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/pwm/Kconfig.intel_blinky [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+config PWM_INTEL_BLINKY
+	bool "Blinky PWM driver"
+	default y
+	depends on DT_HAS_INTEL_BLINKY_PWM_ENABLED
+	help
+	  Enable the INTEL PCH PWM driver found on Intel SoCs
+
+config PWM_INTEL_CONTROL_OFFSET
+	hex "PWM intel control register offset"
+	default 0x304
```
[the remark this answers is not in the document]
Comment (akanisetti) [reply drivers/pwm/Kconfig.intel_blinky:15]: If I add this offset in devicetree I am not getting correct virtual address through MMIO get. So, I added this here.

File: dts/bindings/pwm/intel,blinky-pwm.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: INTEL BLINKY PWM
```
[the remark this answers is not in the document]
Comment (akanisetti) [reply dts/bindings/pwm/intel,blinky-pwm.yaml:5]: No, I will change this to lower case.

File: dts/bindings/pwm/intel,blinky-pwm.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+include: [pwm-controller.yaml, base.yaml]
+
+properties:
+  reg:
+    required: true
+
+  clock-frequency:
+    type: int
+    required: true
+    description: PWM Peripheral Clock frequency in Hz
```
[the remark this answers is not in the document]
Comment (akanisetti) [reply dts/bindings/pwm/intel,blinky-pwm.yaml:18]: clocks is of type "phandle-array" but here I just need to mention IP frequency, so this clock frequency is needed.

**Review by jhedberg — DISMISSED** [review]

File: drivers/pwm/Kconfig.intel_blinky (hunk 3 above)
[the remark this answers is not in the document]
Comment (akanisetti) [reply drivers/pwm/Kconfig.intel_blinky]: Hi @henrikbrixandersen, I have added "reg-offset" property to send it through device tree.

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by finikorg — DISMISSED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

File: tests/drivers/pwm/pwm_api/src/test_pwm.c [hunk 6]
```diff
@@ -58,6 +61,11 @@
 #define DEFAULT_PULSE_CYCLE 512
 #define DEFAULT_PERIOD_NSEC 2000000
 #define DEFAULT_PULSE_NSEC 500000
+#elif DT_HAS_COMPAT_STATUS_OKAY(intel_blinky_pwm)
+#define DEFAULT_PERIOD_CYCLE 32786
+#define DEFAULT_PULSE_CYCLE 16393
```
[the remark this answers is not in the document]
Comment (akanisetti) [reply tests/drivers/pwm/pwm_api/src/test_pwm.c]: Updated, thank you.

**Review by anangl — APPROVED** [review]

**Pull request merged**